### ISSUE TITLE
[google_maps_flutter][Android] Custom marker using BitmapDescriptor render issue

### ISSUE THREAD

**Issue by gabrielginter** [flutter/flutter#141556]
### Steps to reproduce

Only affecting Android, it seems to only affect the app with the Impeller flag enabled but I haven't been able to test it enough to tell for sure.
Unfortunately I can't just disable Impeller as it fixes another issue with Skia in a more critical part of the app.

Markers render incorrectly the first time you create a map instance but works fine with any following map creation.

DO NOT TRY TO REPLICATE IN EMULATOR: BitmapDescriptor markers seem to only work on physical devices (but that's another issue). In the emulator with Impeller ON Markers are rendered like
<details open>
<img width="200" alt="Screenshot 2024-01-15 at 16 22 17" src="https://github.com/flutter/flutter/assets/13461933/9642be3f-a6f8-4eb8-9e85-db556baf11a8">
</details>

---
**Steps to reproduce:**
1. Use a physical android device to replicate
2. Impeller flag enabled
3. Create a simple page with a button navigating to a route with a map, have that map add some markers with a custom icon using BitmapDescriptor
4. Navigate to map
5. Navigate back
6. Navigate to the map once again

---
I will try to add more details once I can do more testing.

### Expected results

Markers render correctly both times the user navigates to the map.

### Actual results

Markers render incorrectly (see image) the first time the user navigates to the map and when navigating for the second time the marker renders correctly.

### Code sample

<details open><summary>Code sample</summary>

I'll add soon

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

First time render
![Screenshot_20240112-144615~2](https://github.com/flutter/flutter/assets/13461933/9332ad98-cf32-4173-b444-fe4634084384)

Second time render
![Screenshot_20240112-144628~2](https://github.com/flutter/flutter/assets/13461933/cbbbaca8-3e51-4b09-b1b0-1fb4fd77db8d)

</details>

### Logs

<details open><summary>Logs</summary>

I don't have access to my physical device right now and the emulator logs are misleading, will add this soon.

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.5, on macOS 14.2.1 23C71 darwin-arm64, locale en-AU)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.85.1)
[✓] Connected device (3 available)
[✓] Network resources

```

</details>

**Comment by dam-ease** [comment]
Thanks for filing this @gabrielginter 
Please can you provide a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that doesn’t include 3rd party plugins or complex production code?

> I'll add soon

Kindly comment once this is added
Thank you

**Comment by gabrielginter** [comment]
Thanks for your reply @dam-ease I have created a minimal app that can reproduce the error:

[https://github.com/gabrielginter/bug_report_141556](https://github.com/gabrielginter/bug_report_141556)

Once you pull the repo, make sure you add a valid API key for google maps in `AndroidManifest.xml` line 7.

I can confirm the issue only happens when Impeller flag is `true`.

**Comment by dam-ease** [comment]
Thanks for your response @gabrielginter.
I can reproduce this issue with Impeller on the latest `stable` channel after about four tries but can't seem to reproduce this on `master`, are you able to check this as it might have been fixed in newer versions?

<details><summary>master flutter doctor -v</summary>
<p>

```
[!] Flutter (Channel master, 3.19.0-6.0.pre.21, on macOS 14.2.1 23C71 darwin-arm64, locale en-NG)
    • Flutter version 3.19.0-6.0.pre.21 on channel master at /Users/damilolaalimi/fvm/versions/master
    ! Warning: `flutter` on your path resolves to /Users/damilolaalimi/sdks/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/damilolaalimi/fvm/versions/master. Consider adding /Users/damilolaalimi/fvm/versions/master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/damilolaalimi/fvm/versions/master. Consider adding /Users/damilolaalimi/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 65acd0728c (3 hours ago), 2024-01-16 23:34:47 -0500
    • Engine revision 73a2de5da5
    • Dart version 3.4.0 (build 3.4.0-24.0.dev)
    • DevTools version 2.31.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C65
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] Connected device (6 available)
    • TECNO CG6 (mobile)          • 067062514C104681                     • android-arm64  • Android 12 (API 31)
    • sdk gphone64 arm64 (mobile) • emulator-5554                        • android-arm64  • Android 14 (API 34) (emulator)
    • Damilola’s iPhone (mobile)  • 00008110-001964480AE1801E            • ios            • iOS 17.1.1 21B91
    • iPhone 15 (mobile)          • DBCC5388-DA57-470E-8F0E-477EA0AA3C3A • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-2 (simulator)
    • macOS (desktop)             • macos                                • darwin-arm64   • macOS 14.2.1 23C71 darwin-arm64
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome 120.0.6099.216

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 

</p>
</details>

**Comment by gabrielginter** [comment]
thanks @dam-ease glad to hear is working on `master`, I am on holidays, I will test this in a week and recommend to close it if it's successful
